tests/generative/data_setup.py

```diff
@@ -81,3 +96,34 @@ def _backend_column(name, type_):
 
 def _backend_table(name, columns):
     return backends.MappedTable(implements=None, source=name, columns=columns)
+
+
+def _build_query(patient_tables, event_tables):
+    # Note that we do not include a schema for the tables here. This is because we are accessing the patient_id
+    # which isn't included in the schema for simplicity elsewhere. There is no downside to not having the schema
+    # because these queries don't mention any other columns, so we don't miss out on possible type-checking.
```

Comment:
I think this reveals something that's been bothering me about the `~patient_id.is_null()` approach which is that `patient_id` doesn't (and _shouldn't_) exist from ehrQL's point of view. And the only reason it works is because we're not using explicit schemas and so the query model just treats it as a vanilla column with type `Any`.

I think the right solution here to support calling `exists_for_patient()` on patient-level frames and then we can drop these references to `patient_id`.